[WIP] CLARIAH Requirements for infrastructure and Software/Services (#5)
CLARIAH/clariah-plus: docs/requirements/infrastructure-requirements.md

CLARIAH aims to develop a service-oriented infrastructure consisting of a wide variety of different applications, this
is often denotes as CLARIAH-as-a-Service (CLaaS) for short.

The user adoption of applications depends not only on their usefulness and quality,
but also on their reliability and performance.
When applications are slow or buggy, users will drop out.
For foundational services such as those delivered by CLARIAH, on which large groups of users depend,
this problem is even more prominent.

Infrastructure plays an important role in delivering reliable software.
This document describes requirements for a fault tolerant, performant and reliable infrastructure.
The requirements are based on modern DevOps and other best practices, including [Twelve-Factor](https://12factor.net)
and [FAIR4RS](https://www.rd-alliance.org/system/files/FAIR4RS_Principles_v0.3_RDA-RFC.pdf).
The [current NDE infrastructure](https://github.com/netwerk-digitaal-erfgoed/infrastructure) is based on these requirements and CLARIAH would do good to follow suit here.

> This document is aimed at organisations and the IT operators within it that would like to take ownership of (‘borgen’) CLARIAH applications.

Reviewer: ```suggestion
This document is aimed at organisations and their IT operators that would like to take ownership of (‘borgen’) CLARIAH applications.
```

The term ‘borgen’ may be too NDE-specific, so I suggest removing it here. Perhaps: ‘that would like to provide CLaaS components’?

### REL

The infrastructure is reliable, which helps adoption of the applications.
Both software developers and end-users need reliable and performant services to get their work done.

#### Example

An infrastructure provider hosts many applications on a single virtual machine (VM).
Application usage increases, and so does server load.
To increase the VM’s resources, the system administrator takes down the VM and deploys a larger instance.
During that time the applications are unavailable to users.
This violates the REL principle.
Horizontal scaling, by running multiple application containers in parallel, would solve this issue.

## Definitions

<dl>
    <dt>Infrastructure</dt>
    <dd>The set of components that enable serving applications to users.
      Functions include not just an execution environment but also code repositories, continuous integration, deployment and monitoring.
      These functions may be delivered by a single component or be spread over multiple components (for instance, GitHub Actions as a CI pipeline, hosting on a cluster).
    </dd>
    <dt>Application</dt>

>     <dd>Any software application developed by Netwerk Digitaal Erfgoed (NDE) including supporting services such as databases.</dd>

Reviewer: ```suggestion
    <dd>Any software application running on CLaaS including supporting services such as databases.</dd>
```
